IdentityCore/src/cache/accessor/MSIDLegacyTokenCacheAccessor.m

```diff
@@ -342,6 +342,46 @@ - (NSArray *)getAllTokensOfType:(MSIDTokenType)tokenType
     return results;
 }
 
+- (BOOL)removeAccount:(MSIDAccount *)account
+              context:(id<MSIDRequestContext>)context
+                error:(NSError **)error
+{
+    // We don't suppport account in legacy cache.
+    return YES;
+}
+
+- (NSArray<MSIDBaseToken *> *)allTokensWithContext:(id<MSIDRequestContext>)context
+                                             error:(NSError **)error
+{
+    MSIDTokenCacheKey *key = [MSIDTokenCacheKey keyForAllItems];
+    __auto_type items = [_dataSource tokensWithKey:key serializer:_serializer context:context error:error];
+    
+    NSMutableArray<MSIDBaseToken *> *tokens = [NSMutableArray new];
+    
+    for (MSIDTokenCacheItem *item in items)
+    {
+        MSIDBaseToken *token = [item tokenWithType:item.tokenType];
+        if (token)
+        {
+            [tokens addObject:token];
+        }
+    }
+    
+    return tokens;
+}
+
+- (BOOL)removeAllTokensForAccount:(MSIDAccount *)account context:(id<MSIDRequestContext>)context error:(NSError *__autoreleasing *)error
+{
+    MSIDLegacyTokenCacheKey *key = [MSIDLegacyTokenCacheKey queryWithAuthority:nil clientId:nil resource:nil legacyUserId:account.legacyUserId];
+    
+    return [_dataSource removeItemsWithKey:key context:context error:error];
+}
+
+- (BOOL)clearWithContext:(id<MSIDRequestContext>)context error:(NSError **)error
+{
+    return [_dataSource removeItemsWithKey:[MSIDTokenCacheKey keyForAllItems] context:nil error:error];
```

Comment:
shouldn't keyForAllitems be called queryForAllItems?

Reply:
renamed